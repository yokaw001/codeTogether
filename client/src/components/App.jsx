import React from "react";

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.bar = this.bar.bind(this);
  }
  bar() {
    return 123;
  }
  componentDidMount() {}
  render() {
    return <div className="barContainer">{() => this.bar()}</div>;
  }
}

export default App;
