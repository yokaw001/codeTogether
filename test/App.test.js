import React from "react";
import { shallow } from "./jestSetup.js";
import App from "../client/src/components/App.jsx";

describe("Loads on startup", () => {
  it("App renders without any issues", () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});

describe("Check helper function", () => {
  it("Helper function returns expected output", () => {
    const component = shallow(<App />);
    const Foo = component.instance().bar();
    expect(Foo).toBe(123);
  });
});

class Foo {
  bar() {
    return 123;
  }
}
describe("Demonstrate Foo is working", () => {
  it("Foo returns expected value", () => {
    expect(new Foo().bar()).toBe(123);
  });
});
