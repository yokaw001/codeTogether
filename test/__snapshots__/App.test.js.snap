// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loads on startup App renders without any issues 1`] = `
<div
  className="barContainer"
>
  <Component />
</div>
`;
